src/browser/components/MainPage.jsx

```diff
@@ -836,6 +839,9 @@ export default class MainPage extends React.Component {
           onLogin={this.handleLogin}
           onCancel={this.handleLoginCancel}
         />
+        <PermissionModal
+          webContentsId={remote.getCurrentWindow().webContents.id}
```

Comment:
Can the `PermissionModal` not get this itself? Doesn't feel right passing props into a component not used by the parent.

Reply:
done, I also realised that we are currently downloading a shim for object.fromEntries already, so i removed the custom code.